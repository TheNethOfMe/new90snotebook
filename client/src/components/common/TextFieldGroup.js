import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

const TextFieldGroup = ({
  name,
  placeholder,
  value,
  label,
  type,
  error,
  info,
  onChange,
  disabled
}) => {
  return (
    <div className="input">
      <label className="input__label" htmlFor={name}>
        {label}
      </label>
      <input
        type={type}
        className={classnames("input__field", {
          "input__invalid-input": error
        })}
        placeholder={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        disabled={disabled}
      />
      {info && <p className="input__info">{info}</p>}
      {error && <p className="input__error">{error}</p>}
    </div>
  );
};

TextFieldGroup.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired
};

TextFieldGroup.defaultProps = {
  type: "text",
  disabled: false
};

export default TextFieldGroup;
