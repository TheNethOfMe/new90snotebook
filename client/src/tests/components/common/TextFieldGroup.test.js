import React from "react";
import { shallow } from "enzyme";
import TextFieldGroup from "../../../components/common/TextFieldGroup";

let wrapper, onChange, error, info;

beforeEach(() => {
  onChange = jest.fn();
  wrapper = shallow(
    <TextFieldGroup
      name="email"
      placeholder="email"
      value=""
      label="Email"
      error={error}
      onChange={onChange}
      info={info}
    />
  );
});

test("should render the text field", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should fire onChange when text is entered", () => {
  wrapper
    .find("input")
    .at(0)
    .simulate("change", "Typing");
  expect(onChange).toHaveBeenCalledWith("Typing");
});
