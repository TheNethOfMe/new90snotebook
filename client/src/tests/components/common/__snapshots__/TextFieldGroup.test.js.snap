// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text field 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextFieldGroup
    disabled={false}
    label="Email"
    name="email"
    onChange={[MockFunction]}
    placeholder="email"
    type="text"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="email"
        >
          Email
        </label>,
        <input
          className=""
          disabled={false}
          name="email"
          onChange={[MockFunction]}
          placeholder="email"
          type="text"
          value=""
        />,
        undefined,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email",
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": "Email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "",
          "disabled": false,
          "name": "email",
          "onChange": [MockFunction],
          "placeholder": "email",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="email"
          >
            Email
          </label>,
          <input
            className=""
            disabled={false}
            name="email"
            onChange={[MockFunction]}
            placeholder="email"
            type="text"
            value=""
          />,
          undefined,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email",
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": "Email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "",
            "disabled": false,
            "name": "email",
            "onChange": [MockFunction],
            "placeholder": "email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
